src/box/sql/where.c

```diff
@@ -4059,22 +4057,17 @@ where_loop_builder_shortcut(struct WhereLoopBuilder *builder)
 	} else {
 		assert(loop->aLTermSpace == loop->aLTerm);
 		struct space *space = item->space;
-		if (space != NULL) {
-			for (uint32_t i = 0; i < space->index_count; ++i) {
-				struct index_def *idx_def =
-					space->index[i]->def;
-				if (!idx_def->opts.is_unique)
-					continue;
-				if (where_loop_assign_terms(loop, clause,
-							    cursor, space_def,
-							    idx_def) == 0)
-					break;
-			}
-		} else {
-			/* Space is ephemeral. */
-			assert(space_def->id == 0);
```

Comment:
This line looks rather strange. I looked at earlier commits and found that `space` was first obtained using [`space_cache_find()`](https://github.com/tarantool/tarantool/commit/a40287051f4d80851c770bfb3d160b85d94f193f) which would return `NULL` for ephemeral spaces. This was later changed to the [current version](https://github.com/tarantool/tarantool/commit/ff75878ab235119bfa067fbf342ceb5773203492), however the new version does not properly check `space` for being ephemeral. So most likely the code is not working as intended. I think a more appropriate solution would be to check if the `space->def->id` is `0` instead of a `space! = NULL`. However, all the mentioned changes happened 5 years ago, so it is entirely possible that the original design will not be applicable. It should probably be investigated.